text/0756-helper-default-manager.md

```diff
@@ -156,6 +176,57 @@ setHelperManager(() => DEFAULT_HELPER_MANAGER, Function.prototype);
  - to register this helper manager, it should occur during app boot so developers do not need to import anything to
    trigger the `setHelperManager` call
 
+**Update 2022-01-13, during implementation**
+During the implementation of this feature, https://github.com/glimmerjs/glimmer-vm/pull/1348
+an issue came up where this RFC doesn't _explicitly_ say what to do with undefined arguments and default arguments.
+
+For example, it was originally assumed that the _last arg_ passed to the helper-function would always be the
+named arguments -- "options-style". This is problematic for positional-only APIs where someone may want to do
+```hbs
+{{myHelper 1 2}}
+or
+{{myHelper 1}}
+```
+where `myHelper` is defined as
+```js
+function myHelper(a, b = 3) {
+  return a * b;
+}
+```
+_with only_ positional arguments, this works as expected, but if named arguments get involved,
+```hbs
+{{myHelper 1 optionA=true}}
+```
+then the named args are passed in place of `b`, which is expected to be a number.
```

Comment:
(This is not quite right with the original proposal)

```suggestion
With the original approach, the first invocation would work as intended, but the second invocation
result in `NaN` unexpectedly. (This is because the function would be called as `myHelper(1, {})`.)

For this reason, we propose to amend the RFC such that the options argument is only passed if
there is at least one named argument in the invocation. In the example above, the function will be
called as `myHelper(1, 2)` and `myHelper(1)`, respectively.
```